### PR TITLE
Programming languages - "Racket" Complete Resources

### LINKED ISSUE #1155: Programming languages - Add complete Racket resources
### Idea Contribution

- [X] I have read all the resource request issues.
- [X] I'm interested in working on this issue
- [X] I'm part of GSSOC organization

### Where do you want to add the resource? Specify the category and section.

Racket is a versatile and powerful language ideal for learning functional programming, creating complex applications, and exploring language design.

I would like to add the complete the complete resources in programming languages

### Provide the URLs of the resources you are adding

_No response_

### Any alternative approaches/features

_No response_

### Additional Context

_No response_

### Code of Conduct

- [X] I follow the [CONTRIBUTING GUIDELINE](https://github.com/jfmartinz/ResourceHub/blob/main/CONTRIBUTING.md) of this project.

### PR DESCRIPTION
**Pull request by aditya-bhaumik**
# Pull Request

## Description
@jfmartinz I have added the complete resources for Racket programming language please review and merge this pull request

Closes #1155  

## Type of change
- [x] Adding a new Resource
- [ ] Adding a new Category
- [ ] Documentation enhancement or fixes
- [ ] Website enhancement or fixes

## Checklist:
- [x] I read carefully [CONTRIBUTING.md](https://github.com/jfmartinz/ResourceHub/blob/main/CONTRIBUTING.md)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: Programming Languages/Racket/README.md [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+   - [File I/O and Networking](#file-io-and-networking)<br>
+   - [Web Development](#web-development)<br>
+- [Resources](#resources)<br> 
+- [Courses](#courses)
+- [Conclusion](#conclusion)
+
+
+## Roadmap
+
+Racket roadmap outlines the essential steps and skills needed to become proficient in Racket programming.
```
Comment (jfmartinz) [inline Programming Languages/Racket/README.md:27]:
```suggestion
> Racket roadmap outlines the essential steps and skills needed to become proficient in Racket programming.
```

File: Programming Languages/Racket/README.md [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+  <tr>
+    <td><a href="https://racket.discourse.group/t/what-is-the-racket-roadmap/2204/6">Racket Roadmap</a></td>
+    <td>A guide to the Racket programming language and its ecosystem.</td>
+  </tr>
+</table>
+
+## Racket
+
+Offers step-by-step guides and tutorials covering various aspects of Racket programming. Helps beginners understand fundamental concepts and advanced users explore new techniques.
+Introduction
```
Comment (jfmartinz) [inline Programming Languages/Racket/README.md:42]:
```suggestion
> Offers step-by-step guides and tutorials covering various aspects of Racket programming. Helps beginners understand fundamental concepts and advanced users explore new techniques.
Introduction
```

File: Programming Languages/Racket/README.md [hunk 3]
```diff
@@ -0,0 +1,270 @@
+## Racket 🎨
```
Comment (jfmartinz) [inline Programming Languages/Racket/README.md:1]:
```suggestion
# Racket 
```

File: Programming Languages/Racket/README.md [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+    <td><a href="https://medium.com/@G_Kumar/racket-programming-language-features-c71b83b625c4">Feature</a></td>
+    <td>Learn about the features of Racket programming.</td>
+  </tr>
+  <tr>
+    <td><a href="https://docs.racket-lang.org/quick/">Quick Introduction</a></td>
+    <td>Racket provides a quick introduction to its syntax and features.</td>
+  </tr>
+</table>
+
+## Basics
```
Comment (jfmartinz) [inline Programming Languages/Racket/README.md:62]:
```suggestion
### Basics
```

File: Programming Languages/Racket/README.md [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+    <td><a href="https://docs.racket-lang.org/guide/datatypes.html">Data Types</a></td>
+    <td>An overview of the various data types available in Racket.</td>
+  </tr>
+  <tr>
+    <td><a href="https://home.adelphi.edu/~siegfried/cs270/270rl6.html">Syntax and Semantics</a></td>
+    <td>Learn the syntax rules and semantic conventions of Racket.</td>
+  </tr>
+</table>
+
+## Functions
```
Comment (jfmartinz) [inline Programming Languages/Racket/README.md:87]:
```suggestion
### Functions
```

File: Programming Languages/Racket/README.md [hunk 6]
```diff
@@ -0,0 +1,270 @@
...
+    <td><a href="https://docs.racket-lang.org/mischief/function.html">Higher-Order Functions</a></td>
+    <td>Explore the concept of higher-order functions in Racket.</td>
+  </tr>
+  <tr>
+    <td><a href="https://docs.racket-lang.org/guide/Lists__Iteration__and_Recursion.html">Recursion</a></td>
+    <td>An article on using recursion in Racket programming.</td>
+  </tr>
+</table>
+
+## Data Structures
```
Comment (jfmartinz) [inline Programming Languages/Racket/README.md:108]:
```suggestion
### Data Structures
```

File: Programming Languages/Racket/README.md [hunk 7]
```diff
@@ -0,0 +1,270 @@
...
+    <td><a href="https://docs.racket-lang.org/reference/vectors.html">Vectors</a></td>
+    <td>This article will look at what vectors are and how they are used.</td>
+  </tr>
+  <tr>
+    <td><a href="https://docs.racket-lang.org/reference/hashtables.html">Hash Tables</a></td>
+    <td>An overview of hash tables in Racket and their uses.</td>
+  </tr>
+</table>
+
+## Macros
```
Comment (jfmartinz) [inline Programming Languages/Racket/README.md:129]:
```suggestion
### Macros
```

File: Programming Languages/Racket/README.md [hunk 8]
```diff
@@ -0,0 +1,270 @@
...
+    <td><a href="https://docs.racket-lang.org/guide/pattern-macros.html#:~:text=The%20define%2Dsyntax%2Drule%20form,a%20use%20of%20the%20macro.">Syntax Rules</a></td>
+    <td>An overview of defining macros using syntax rules.</td>
+  </tr>
+  <tr>
+    <td><a href="https://download.racket-lang.org/docs/5.1/html/macro-debugger/index.html">Macro Debugging</a></td>
+    <td>Learn techniques for debugging macros in Racket.</td>
+  </tr>
+</table>
+
+## Concurrency
```
Comment (jfmartinz) [inline Programming Languages/Racket/README.md:149]:
```suggestion
### Concurrency
```

File: Programming Languages/Racket/README.md [hunk 9]
```diff
@@ -0,0 +1,270 @@
...
+    <td><a href="https://docs.racket-lang.org/reference/threads.html">Threads</a></td>
+    <td>An article on using threads for concurrent programming in Racket.</td>
+  </tr>
+  <tr>
+    <td><a href="https://docs.racket-lang.org/reference/concurrency.html">Concurrency Primitives</a></td>
+    <td>An overview of concurrency primitives available in Racket.</td>
+  </tr>
+</table>
+
+## Modules and Libraries
```
Comment (jfmartinz) [inline Programming Languages/Racket/README.md:166]:
```suggestion
### Modules and Libraries
```

File: Programming Languages/Racket/README.md [hunk 10]
```diff
@@ -0,0 +1,270 @@
...
+    <td><a href="https://docs.racket-lang.org/guide/modules.html">Modules</a></td>
+    <td>Learn about modules in Racket for organizing code.</td>
+  </tr>
+  <tr>
+    <td><a href="https://docs.racket-lang.org/pkg/index.html">Package System</a></td>
+    <td>An overview of Racket's package system for managing libraries.</td>
+  </tr>
+</table>
+
+## Error Handling
```
Comment (jfmartinz) [inline Programming Languages/Racket/README.md:182]:
```suggestion
### Error Handling
```

File: Programming Languages/Racket/README.md [hunk 11]
```diff
@@ -0,0 +1,270 @@
...
+    <td><a href="https://docs.racket-lang.org/reference/exns.html">Exception Handling</a></td>
+    <td>An article on handling exceptions in Racket.</td>
+  </tr>
+  <tr>
+    <td><a href="https://beautifulracket.com/explainer/errors-and-exceptions.html">Built-in Exceptions</a></td>
+    <td>An overview of built-in exceptions in Racket.</td>
+  </tr>
+</table>
+
+## File I/O and Networking
```
Comment (jfmartinz) [inline Programming Languages/Racket/README.md:198]:
```suggestion
### File I/O and Networking
```

File: Programming Languages/Racket/README.md [hunk 12]
```diff
@@ -0,0 +1,270 @@
...
+    <td><a href="https://docs.racket-lang.org/reference/Filesystem.html">File I/O</a></td>
+    <td>Learn how to handle file input and output in Racket.</td>
+  </tr>
+  <tr>
+    <td><a href="https://docs.racket-lang.org/reference/networking.html">Networking</a></td>
+    <td>An article on networking capabilities in Racket.</td>
+  </tr>
+</table>
+
+## Web Development
```
Comment (jfmartinz) [inline Programming Languages/Racket/README.md:214]:
```suggestion
### Web Development
```

File: Programming Languages/Racket/README.md [hunk 13]
```diff
@@ -0,0 +1,270 @@
...
+  </tr>
+  <tr>
+    <td><a href="https://github.com/greghendershott/racket-mode">Emacs Racket Mode</a></td>
+    <td>This project adds Racket support to Emacs.</td>
+  </tr>
+</table>
+
+## Courses
+
+Here are courses you may consider:
```
Comment (jfmartinz) [inline Programming Languages/Racket/README.md:252]:
```suggestion
> Here are courses you may consider:
```

File: Programming Languages/Racket/README.md [hunk 14]
```diff
@@ -0,0 +1,270 @@
...
+    <td><a href="https://docs.racket-lang.org/web-server/run.html">Servlets</a></td>
+    <td>Learn about creating and using servlets in Racket web applications.</td>
+  </tr>
+  <tr>
+    <td><a href="https://docs.racket-lang.org/json/index.html#:~:text=3%20Parsing%20JSON%20Text%20into%20JS%2DExpressions,-Link%20to%20this&text=Reads%20a%20jsexpr%20from%20a,remaining%20JSON%20input(s).">JSON Handling</a></td>
+    <td>An overview of working with JSON in Racket.</td>
+  </tr>
+</table>
+
+## Resources
```
Comment (jfmartinz) [inline Programming Languages/Racket/README.md:234]:
```suggestion
## Resources
```

Add the description

**Review by jfmartinz — CHANGES_REQUESTED** [review]

**Comment by aditya-bhaumik** [comment]
@jfmartinz i made the changes requested please review and merge this pull request

**Comment by jfmartinz** [comment]
You forgot Module Libraries. Make it h3

**Comment by aditya-bhaumik** [comment]
@jfmartinz done

**Review by jfmartinz — APPROVED** [review]

**Comment by aditya-bhaumik** [comment]
@jfmartinz please add the level label

**Pull request merged**